frontend/src/SwimMeet/SwimMeetDisplay.jsx

```diff
@@ -63,26 +70,50 @@ const SwimMeetDisplay = () => {
   };
 
   const actions = [
-    { name: "Details", icon: <DetailsIcon />, onClick: handleDetailsClick, tip:"Go to events" },
-    { name: "Edit", icon: <EditIcon />, onClick: handleEditClick, tip:"Edit basic information" },
-    { name: "Delete", icon: <DeleteIcon />, onClick: handleDeleteClick, tip: "Delete" },
-    { name: "Ranking", icon: <RankingIcon />, onClick: handleRankingClick, tip: "Go to ranking" },
+    {
+      name: "Details",
+      icon: <DetailsIcon />,
+      onClick: handleDetailsClick,
+      tip: "Go to events",
+    },
+    {
+      name: "Edit",
+      icon: <EditIcon />,
+      onClick: handleEditClick,
+      tip: "Edit basic information",
+    },
+    {
+      name: "Delete",
+      icon: <DeleteIcon />,
+      onClick: handleDeleteClick,
+      tip: "Delete",
+    },
+    {
+      name: "Ranking",
+      icon: <RankingIcon />,
+      onClick: handleRankingClick,
+      tip: "Go to ranking",
+    },
   ];
 
   useEffect(() => {
     let ignore = false;
     async function fetching() {
-      const json = await SmmApi.getSwimMeetList(searchPar, offset, limit);
-      console.log(json);
-      if (!ignore) {
-        const formattedData = json.results.map(item => ({
-          ...item,
-          date: dayjs(item.date).format('MM/DD/YYYY'),
-          time: item.time.slice(0, 5),
-        }));
+      try {
+        const json = await SmmApi.getSwimMeetList(searchPar, offset, limit);
+        console.log(json);
+        if (!ignore) {
+          const formattedData = json.results.map((item) => ({
+            ...item,
+            date: dayjs(item.date).format("MM/DD/YYYY"),
+            time: item.time.slice(0, 5),
+          }));
 
-        setData(formattedData);
-        setCount(json.count)
+          setData(formattedData);
+          setCount(json.count);
+        }
+      } catch (error) {
+        setErrorOnLoading(true);
```

Comment:
Once `errorOnLoading` is set to `true`, it stays that way. After reading the data, `errorOnLoading` should be set to `false`.

In `AddSwimMeet`, we did not set `errorOnLoading` to false after reading the data because the `useEffect` is only used on the first render.